lib/mindee/client.rb

```diff
@@ -62,13 +62,15 @@ def parse(
     )
 
       @doc_config = find_doc_config(prediction_class, endpoint_name, account_name)
-      input_doc.process_pdf(page_options) if !page_options.nil? && input_doc.pdf?
+      if !page_options.nil? && input_doc.pdf? && input_doc.is_a?(Mindee::Input::LocalInputSource)
```

Comment:
minor comment, by putting this first we save some processing

```suggestion
      if input_doc.is_a?(Mindee::Input::LocalInputSource) && !page_options.nil? && input_doc.pdf?
```